mend_sca_cleanup_tool/sca_cleanup_tool.py

```diff
@@ -58,35 +59,63 @@ def main():
         CONFIG = parse_args()
         
     setup_config()
+    if CONFIG.skip_report_generation:
+        print("Skip Report Generation Enabled - Reports will not be generated")
+    if CONFIG.skip_project_deletion:
+        print("Skip Project Deletion Enabled - Projects will not be deleted")
 
     if CONFIG.dry_run:
         print("Dry Run enabled - no reports or deletions will occur")
 
-    product_project_dict = get_projects_to_remove()
-    total_projects_to_delete = (sum([len(product_project_dict[x]) for x in product_project_dict]))
-    if not CONFIG.dry_run:
-        if total_projects_to_delete == 0:
-            print("No projects to clean up were found")
-            exit()
-        else:
-            print(f"Found {total_projects_to_delete} project(s) to delete, generating reports and removing project(s)...")
-        for product_token in product_project_dict:
-            for project in product_project_dict[product_token]:
-                if not CONFIG.skip_report_generation:
-                    try:
-                        generate_reports(project)
-                    except:
-                        print(f"There was an issue with the report generation, skipping deletion for project: {project['name']}")
-                        continue
-                else:
-                    print("skipReportGeneration flag found, skipping report generation")
-                if not CONFIG.skip_project_deletion:
-                    delete_scan(product_token, project)
-                else:
-                    print("skipProjectDeletion flag found, skipping project deletion")
+    processed_projects = {}
+    products = get_products()
+    with ThreadPoolExecutor(max_workers=CONFIG.project_parallelism_level) as executor:
```

Comment:
``CONFIG.project_parallelism_level`` gets injested from params.config as a string. This needs to be casted properly to an int.